app/models/info_request.rb

```diff
@@ -1319,11 +1319,11 @@ def initial_request_text
 
   # Returns index of last event which is described or nil if none described.
   def index_of_last_described_event
-    events = info_request_events
-    events.each_index do |i|
-      revi = events.size - 1 - i
-      m = events[revi]
-      return revi if m.described_state
+    info_request_events.reverse.each_with_index do |event, index|
+      if event.described_state
+        reverse_index = info_request_events.size - 1 - index
+        return reverse_index
```

Comment:
On second look _could_ just return the calculation here but I don't find the purpose of the calculation immediately obvious. An alternative - with a presumed tiny efficiency saving over assigning a variable - would be to simply add a comment above `return info_request_events.size - 1 - index` instead but I quite like that the self-describing code